drivers/ethernet/Kconfig.mcux

```diff
@@ -89,30 +77,6 @@ config ETH_MCUX_0_MANUAL_MAC
 
 endchoice
 
-if ETH_MCUX_0_MANUAL_MAC
```

Comment:
Why removing the mac address from Kconfig? If one wants to have a fixed MAC address, then one is forced to edit a file in version control which is not very convenient, or did I understood something wrong here.

Reply:
@jukkar The MAC is currently set in Kconfig to UNIQUE mode, i.e. implicit value. Manual MAC configured in DT is used only if Kconfig has set ETH_MCUX_0_MANUAL_MAC.

Reply:
Sure, my point was that if one wants to set the MAC address, then one needs to edit the DT file which is under version control instead of putting the MAC address into a config file that is probably not under version control. Or did I misunderstood something here.

Reply:
Kconfig file is also under version control (file in discussion). I was pointing out that if you want to change the MAC manually then you need to make more changes and dt is not enough.

Reply:
> Kconfig file is also under version control (file in discussion)

Sure, Kconfig file is but that does not contain the actual values of the MAC address as that is placed to the prj*conf. User is free to select his own conf file during compilation.

Not sure what do you mean by "...then you need to make more changes and dt is not enough."

Reply:
> Not sure what do you mean by "...then you need to make more changes and dt is not enough."

I was pointing out that with this change MAC should be changed via device tree.

Reply:
@jukkar:

> Sure, my point was that if one wants to set the MAC address, then one needs to edit the DT file which is under version control

I find myself recently quoting @nashif on multiple times who said that we have [DT overlays](https://docs.zephyrproject.org/latest/application/application.html#application-dt). But I yet need to try myself how cumbersome it is. One point to have is that we won't get around need to override DT settings, so the earlier we start get used to it, the sooner we can may it more streamlined (if it's not such already).

I have mixed feelings about switching to `local-mac-address` in DT myself. But it's worth to point that it at least not our invention, it comes from Linux: https://elixir.bootlin.com/linux/latest/source/Documentation/devicetree/bindings/net/ethernet.txt . And if we need to go thru pain of it, then at least one can imagine a benefit: we have quite adhoc configs for manual MACs in various drivers, DT would standardize somewhat.

How much - depends, and we (those of us who don't know DT's ins and out) yet need to see. For example, will it be possible to have device-independent DT overlay which would specify MAC for any ethernet device? (E.g. using "eth" alias, as this patch introduces.)

All in all, I'm going to approve this, as at least it all makes sense and seems to work, while keeping the good defaults we have.

Reply:
Setting things via conf file feels more natural to me that doing things via some overlay DT. Anyway, I have no strong opinion on this and can for sure use some other board if frdm becomes too cumbersome to use in real life.

Reply:
> and can for sure use some other board if frdm becomes too cumbersome to use in real life.

Nah-nah, the whole point of this is that everything gets converted ;-). For example, recently @aescolar complained (bragged?) that native_posix doesn't use DT. I'm sure, it will be converted too.